test/type/runtime_discoverable_attrs_distributed.swift

```diff
@@ -0,0 +1,39 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend -emit-module %S/Inputs/RuntimeAttrs.swift -o %t
+// RUN: %target-typecheck-verify-swift -disable-availability-checking -enable-experimental-feature RuntimeDiscoverableAttrs -I %t
+
+// REQUIRES: asserts
+
+import RuntimeAttrs
+import Distributed
+
+typealias DefaultDistributedActorSystem = LocalTestingDistributedActorSystem
+
+@runtimeMetadata
+struct FlagForAsyncFuncs {
+  init<Act>(attachedTo: (Act) async throws -> Void) {}
+  init<Act>(attachedTo: (Act, Int, inout [Int]) async throws -> Void) {}
+  init<Act>(attachedTo: (Act, Int) async -> Void) {}
+  init(attachedTo: () async -> [String]) {}
+}
+
+@ActorFlag
+distributed actor TestDistributedActor {
+  @FlagForAsyncFuncs distributed func asyncExternallyDist() throws {} // ok
+
+  @FlagForAsyncFuncs distributed func doSomethingDist() async throws {} // ok
+
+  @FlagForAsyncFuncs nonisolated func doSomethingNonisolated(_: Int) async {} // ok
+
+  @FlagForAsyncFuncs distributed func doSomethingDist(_: Int, x: [Int]) async {} // ok
```

Comment:
This seems wrong, the matcher is `inout Int` but our func is not inout but it passed?

Reply:
heh, welcome to our matching, this would actually pick the first overload `<Act>(attachedTo: (Act) async throws -> Void)` because generic parameter is allowed to get bound to a tuple. You have to constraint `Act` to `DistributedActor` for all your overloads for it to match the way you want.

Reply:
Aha, thanks for explaining this all :)